### PR TITLE
[Merged by Bors] - refactor: Names the function of Equiv.piFinSuccAbove

### PR DESCRIPTION
**Pull request by BoltonBailey**
Creates `extractNth` to do the opposite of `insertNth` for tuples. Rewrites the `Equiv` definition.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by BoltonBailey** [comment]
Perhaps some API is in order here too.

**Comment by BoltonBailey** [comment]
Ok, I added one explicit lemma linking the function to insertNth and putting it next to it.

File: Mathlib/Logic/Equiv/Fin.lean
Comment (YaelDillies) [inline Mathlib/Logic/Equiv/Fin.lean:295]:
```suggestion
  left_inv f := by simp
```
since that's what you call it above

**Review by YaelDillies — APPROVED** [review]
This looks fine but I want @linesthatinterlace's approval.

**Comment by linesthatinterlace** [comment]
This is really nice (and a thing I was wanting, though I've been busy with other stuff the last few months). I can't really see any obvious problems with this: there might be some more lemmas you end up needing, of course, but they can be added as needed, and you have the important ones.

**Comment by YaelDillies** [comment]
Okay great! Then LGTM

maintainer merge

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**
